src/msix/pack/AppxPackageWriter.cpp

```diff
@@ -65,11 +105,7 @@ namespace MSIX {
 
     HRESULT STDMETHODCALLTYPE AppxPackageWriter::Close(IStream* manifest) noexcept try
     {
-        ThrowErrorIf(Error::InvalidState, m_state != WriterState::Open, "Invalid package writer state");
-        auto failState = MSIX::scope_exit([this]
-        {
-            this->m_state = WriterState::Failed;
-        });

```

Comment:
This is my bad. We should check for the state but only after checking if its a bundle. Also, something can go wrong while reading the manifest file, so the failState should be released here and not in CloseInternal. Can you take that back CloseInternal? The close of the bundle will need something similar.

Reply:
Reverted this file entirely since I created a separate file for bundles, AppxBundleWriter.cpp.